Week6/MyFavoriteMovies/app/src/main/java/com/raywenderlich/myfavoritemovies/activities/DetailActivity.kt

```diff
@@ -24,9 +28,12 @@ class DetailActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_detail)
-
-        val movie = repository.getMovieById(movieId)
-        displayMovieDetails(movie)
+        lifecycleScope.launch(Dispatchers.IO) {
+            val movie = repository.getMovieById(movieId)
```

Comment:
If `getMovieById()` is a Room-suspend function, then you don't have to switch to `Dispatchers.IO`, and then back to the Main thread, you can just call launch!